Sources/UserStore/UserSessionStub.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  UserSessionStub.swift
+//
+//
+//  Created by Anian Schleyer on 03.06.24.
+//
+
+import Foundation
+import Common
+
+public class UserSessionStub: UserSession {
+    override init() {
```

Comment:
Could we also follow the `protocol`, `Impl`, and `Factory` pattern here?

I'd much prefer that over a sub-class.

Reply:
I tried, but unfortunately this doesn't work: UserSession needs to have `@Published` variables, which can't be part of a protocol (at least I couldn't get it to work). You can only have normal variables, which you then can't override with variables that have a property wrapper :/

Reply:
Yeah, it's not easy. We could inherit the protocol that synthesizes the `objectWillChange` publisher:

```swift
protocol UserSession: ObservableObject {}

enum UserSessionFactory {
    static let shared: any UserSession = UserSessionImpl()
}
```

`any` is not great, but necessary because of the `associatedtype`.